engine-api/src/json_utils.rs

```diff
@@ -30,3 +30,7 @@ pub fn deserialize<T: DeserializeOwned>(x: &serde_json::Value) -> Result<T, Json
 pub fn access_state_error<E: fmt::Debug>(e: E) -> JsonRpcError {
     JsonRpcError::access_state_error(e)
 }
+
+pub fn transaction_error<E: fmt::Debug>(e: E) -> JsonRpcError {
+    JsonRpcError::without_data(32603, format!("Invalid transaction: {e:?}"))
```

Comment:
Is this status code what is returned by geth for this class of errors?

Reply:
I got the error code from Ethereum docs: https://ethereum-json-rpc.com/errors

When I send an invalid transaction to `geth`, I get the following error message:
```json
"error": {
    "code": 3,
    "message": "execution reverted",
    "data": "0x000000000000005e"
}
```

So seems like we can return error code `3` with a static `Execution reverted` message.